src/sentry/processing/realtime_metrics/redis.py

```diff
@@ -77,31 +87,146 @@ def increment_project_duration_counter(
+
+        This may throw an exception if there is some sort of issue scanning the redis store for
+        projects.
+        """
+
+        already_seen = set()
+        # Normally if there's a histogram entry for a project then there should be a counter
+        # entry for it as well, but double check both to be safe
+        all_keys = chain(
+            self.cluster.scan_iter(
+                match=self.counter_key_prefix() + ":*",
+            ),
+            self.cluster.scan_iter(
+                match=self.histogram_key_prefix() + ":*",
+            ),
+        )
+
+        for item in all_keys:
+            # Because this could be one of two patterns, this splits based on the most basic
+            # delimiter ":" instead of splitting on known prefixes
+            _prefix, _metric_type, _bucket_size, project_id_raw, _else = item.split(":", maxsplit=4)
+            project_id = int(project_id_raw)
+            if project_id not in already_seen:
+                already_seen.add(project_id)
+                yield project_id
+
+    def get_counts_for_project(self, project_id: int) -> Iterable[base.BucketedCount]:
+        """
+        Returns a sorted list of bucketed timestamps paired with the count of symbolicator requests
+        made during that time for some given project.
+
+        This may throw an exception if there is some sort of issue fetching counts from the redis
+        store.
+        """
+        key_prefix = f"{self.counter_key_prefix()}:{project_id}:"
+
+        keys = sorted(
+            self.cluster.scan_iter(
+                match=key_prefix + "*",
+            )
+        )
+        counts = self.cluster.mget(keys)
+        for key, count_raw in zip(keys, counts):
+            _, timestamp_raw = key.split(key_prefix)
+
+            try:
+                timestamp_bucket = int(timestamp_raw)
+                count = int(count_raw)
+            except Exception:
+                logger.warning(
+                    'Ignoring symbolicator metric: project %s key %s, timestamp or count could not be converted to int: timestamp "%s" count "%s"',
+                    project_id,
+                    key,
+                    count_raw,
+                )
+            else:
+                yield base.BucketedCount(timestamp=timestamp_bucket, count=count)
+
+    def get_durations_for_project(self, project_id: int) -> Iterable[base.DurationHistogram]:
+        """
+        Returns a sorted list of bucketed timestamps paired with a histogram-like dictionary of
+        symbolication durations made during some timestamp for some given project.
+
+        For a given `{duration:count}` entry in the dictionary bound to a specific `timestamp`:
+
+        - `duration` represents the amount of time it took for a symbolication request to complete.
+        Durations are bucketed by 10secs, meaning that a `duration` of `30` covers all requests that
+        took between 30-39 seconds.
```

Comment:
yep! added this in just in case the user's unfamiliar with the syntax: i would imagine trying to look this up would be a pain if all you knew was "square bracket" and "round bracket"